intl_translation_format/README.md

```diff
@@ -56,6 +56,169 @@ pub run intl_translation_format:generate
 ```
 
 ### Implement a new translation file format
-[TBD]
 
+The intl_translation_format brings the tools needed to support a new translation format. 
+You can extend the class TranslationFormat and complete the functions that will take care of the extracting of the messages
+and create templates. There are several classes to make this task easier: `MonoLingualFormat`, `BinaryMonolingualFormat`,
+`MultiLingualFormat`.
 
+As an example, this guide will saw you how the JSON file format is implemented.
+
+1. Learning about your format
+
+It is important to do some research about the format you are interested to implement. Some good questions would be:
+ 1. Is it a standard? 
+ 2. Is it a binary or text-based format?
+ 3. Does it contain a translation for multiple languages per file or is it just one language per file?
```

Comment:
Maybe these could have examples, and briefly indicate the consequence. e.g. For example, ARB has a separate file for each language which we call a monolingual format. XLIFF can contain multiple languages in a single file, which we call a multilingual format.

Helps clarify what those terms mean.